pkg/frontend/querymiddleware/querysharding_test.go

```diff
@@ -1963,6 +2077,14 @@ func newSeries(metric labels.Labels, from, to time.Time, step time.Duration, gen
 	})
 }
 
+func generateTestHistogram(v float64) *histogram.FloatHistogram {
+	h := util_test.GenerateTestGaugeFloatHistogram(int(v))
```

Comment:
```suggestion
	h := util_test.GenerateTestFloatHistogram(int(v))
```

Due to "increase should only be used with counters and native histograms where the components behave like counters. It is syntactic sugar for rate(v) multiplied by the number of seconds under the specified time range window, and should be used primarily for human readability. Use rate in recording rules so that increases are tracked consistently on a per-second basis." in prometheus docs.

This is not to say that we should avoid splitting such queries - it would be impossible since we don't know beforehand what the values will be. Just to say that we're going to test correctness with counter.

Reply:
Hmm perhaps, I just made it a gauge because the value of v may go up and down, and also sometimes it may be NaN for the stale marker. But either way they both pass tests so I'm not sure if it matters.